samples/azureeventsourcelistener-eventhubs-logging/Program.cs

```diff
@@ -0,0 +1,140 @@
+        /// </remarks>
+        ///
+        public static async Task Main(string[] args)    
+        {
+            var connectionString = default(string);
+            var eventHubName = default(string);
+            var instrumentationKey = default(string);
+            // if we have more than two argument passed we asusme that the first is the connection string , 
+            // the second is the event hub name and the third is intrumentation key for application insights.
+            if (args.Count() > 2)
+            {
+                connectionString = args[0];
+                eventHubName = args[1];
+                instrumentationKey = args[2];
+            }
+
+            // If there was no connection string provided, then prompt for one.
+            while (string.IsNullOrEmpty(connectionString))
+            {
+                Console.Write("Please provide the connection string for the Event Hubs that you'd like to use and then press Enter: ");
+                connectionString = Console.ReadLine().Trim();
+                Console.WriteLine();
+            }
+
+            // If there was no eventHubName provided, then prompt for one.
+            while (string.IsNullOrEmpty(eventHubName))
+            {
+                Console.Write("Please provide the Event Hub Name that you'd like to use and then press Enter: ");
+                eventHubName = Console.ReadLine().Trim();
+                Console.WriteLine();
+            }
+
+            // If there was no instrumentation Key provided, then prompt for one.
+            while (string.IsNullOrEmpty(instrumentationKey))
+            {
+                Console.Write("Please provide the Event Hub Name that you'd like to use and then press Enter: ");
+                instrumentationKey = Console.ReadLine().Trim();
+                Console.WriteLine();
+            }
+
+            //Create ApplicationInsights Client.
+            var telemetryClient = new TelemetryClient(new TelemetryConfiguration(instrumentationKey));
+            
+            //Create EventHub Producer Client.
+            var producerClient = new EventHubProducerClient(connectionString, eventHubName);
+
+            try
+            {
+                // Setup a listener to monitor logged events.
+                var listener = new AzureEventSourceListener((args, message) =>
+                {
+                    // Add condition to manipulate only events related to EventHubs.
+                    if (args.EventSource.Name.StartsWith("Azure-Messaging-EventHubs"))
+                    {
+                        var properties = GetPropertiesDictionnary(args.PayloadNames, args.Payload);
+                        var level = args.Level.GetSeverityLevel();
+                        var traceMessage = string.Format(args.Message, args.Payload.ToArray());
+                        telemetryClient.TrackTrace($"Sample_AzureEventSourceListener | {traceMessage}", level, properties);
```

Comment:
I didn't get it, 
do you talk about the message that we log to app insights ?

Reply:
Yes, that's what I meant.   We're identifying the name of the application, but the message itself is content of the log message.   It may be helpful to add a prefix to indicate that this was captured as part of the Event Hubs client log.   Maybe something like:

```csharp
var traceMessage = $"Event Hubs: { string.Format(CultureInfo.InvariantCulture, args.Message, args.Payload.ToArray()) }";
```